### PR TITLE
docs: how-to guide to control casing (DOCS-4102)

### PR DESCRIPTION
**Pull request by MichaelDrogalis**
### Description 

A new how-to guide to answer a commonly asked question: how do I make ksqlDB stop uppercasing everything?

### CODE REVIEW

File: docs/how-to-guides/control-the-case-of-identifiers.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+struct. Any references to struct fields must exactly match the casing
+ksqlDB expects. ksqlDB will reject any UDF invocations that do not
+match it.
+
+Here is a quick example of using backticks in a UDF that returns a
+custom struct. The first two fields override the default behavior. The
+last uses the default casing. That means that when you work with the
+data returned from the UDF in ksqlDB, all select statements must use
+this exact casing. For more information on working with structs in
+UDFs, see the struct section of the [wow to create a user-defined
```
Comment (JimGalasyn) [inline docs/how-to-guides/control-the-case-of-identifiers.md:94]:
```suggestion
UDFs, see the struct section of the [how to create a user-defined
```

Comment (MichaelDrogalis) [reply docs/how-to-guides/control-the-case-of-identifiers.md:94]: wow 😍

File: docs/how-to-guides/control-the-case-of-identifiers.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    c VARCHAR
+) WITH (
+    kafka_topic = 's2',
+    partitions = 1,
+    value_format = 'avro'
+);
+```
+
+Insert some rows into `s2`. Notice how you need to use backticks each
+time to reference a field that doesn't have the default casing:
```
Comment (agavra) [inline docs/how-to-guides/control-the-case-of-identifiers.md:52]: unless it's all uppercase 😂 you could reference `B` without quotes. might be good to have an example with mixed case (`` `Dog` ``)

File: docs/how-to-guides/control-the-case-of-identifiers.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+```sql
+INSERT INTO `s2` (`a`, `B`, c) VALUES ('k1', 1, 'x');
+INSERT INTO `s2` (`a`, `B`, c) VALUES ('k2', 2, 'y');
```
Comment (agavra) [inline docs/how-to-guides/control-the-case-of-identifiers.md:56]:
```suggestion
INSERT INTO `s2` (`a`, `B`, C) VALUES ('k2', 2, 'y');
```
to illustrate that unquoted cols are case-insensitive

File: docs/how-to-guides/control-the-case-of-identifiers.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+Declare a new stream named `s2`. In this example, you override
+ksqlDB's default behavior to uppercase all identifiers. Use backticks
+to control the casing of the stream name and two of the column names,
+`a` and `B`. `c` receives the default behavior of uppercasing.
+
+```sql
+CREATE STREAM `s2` (
+    `a` VARCHAR KEY,
+    `B` INT,
+    c VARCHAR
```
Comment (agavra) [inline docs/how-to-guides/control-the-case-of-identifiers.md:43]: might be good to have a `struct` example where the fields in the struct are cased (and link it to the example below)

File: docs/how-to-guides/control-the-case-of-identifiers.md (hunk 2 above)
Comment (agavra) [reply docs/how-to-guides/control-the-case-of-identifiers.md:52]: also if you have `` `B` `` and `` `b` `` as two different fields, `b` would resolve to `` `B` `` and `` `b` `` would resolve to `` `b` `` (correctly)

File: docs/how-to-guides/control-the-case-of-identifiers.md (hunk 4 above)
Comment (MichaelDrogalis) [reply docs/how-to-guides/control-the-case-of-identifiers.md:43]: Great call.

File: docs/how-to-guides/control-the-case-of-identifiers.md (hunk 3 above)
Comment (MichaelDrogalis) [reply docs/how-to-guides/control-the-case-of-identifiers.md]: Good suggestion. Handled it differently in the last patch.

File: docs/how-to-guides/control-the-case-of-identifiers.md (hunk 2 above)
Comment (MichaelDrogalis) [reply docs/how-to-guides/control-the-case-of-identifiers.md:58]: I think I covered this without making it too confusing. See what you think.

**Comment by MichaelDrogalis** [comment]
@agavra Added struct example + mixed casing + clarified a few pieces of text. If you could give it one more quick pass for accuracy, that would be super.

File: docs/how-to-guides/control-the-case-of-identifiers.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+in each topic.
+
+```sql
+SET 'auto.offset.reset' = 'earliest';
+```
+
+Declare a new stream named `s2`. In this example, you override
+ksqlDB's default behavior to uppercase all identifiers. Use backticks
+to control the casing of the stream name and column names. To contrast
+this behavior, `quz` is declared without backticks to demonstrate
```
Comment (agavra) [inline docs/how-to-guides/control-the-case-of-identifiers.md:37]:
```suggestion
this behavior, `qux` is declared without backticks to demonstrate
```

File: docs/how-to-guides/control-the-case-of-identifiers.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+the default behavior of uppercasing.
+
+```sql
+CREATE STREAM `s2` (
+    `foo` VARCHAR KEY,
+    `BAR` INT,
+    `Baz` VARCHAR,
+    `grault` STRUCT<
+        `Corge` VARCHAR,
+        `garply` INT
```
Comment (agavra) [inline docs/how-to-guides/control-the-case-of-identifiers.md:47]: out of interest, where did you come up with these? 😂

Comment (agavra) [reply docs/how-to-guides/control-the-case-of-identifiers.md:47]: elementary, my dear Watson: https://en.wikipedia.org/wiki/Metasyntactic_variable

Comment (agavra) [reply docs/how-to-guides/control-the-case-of-identifiers.md:47]: we should add `franz` to our standard canon of metasyntactic variables

Comment (MichaelDrogalis) [reply docs/how-to-guides/control-the-case-of-identifiers.md:47]: Hah, you found it. :) I ran out and googled "foo bar baz qux" and clicked the first link. Glad there's more, my choices would have been terrible. 😂

**Pull request merged**